bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/api/rest/WebSocketManager.java

```diff
@@ -82,25 +95,50 @@ public String getEvent() {
         }
     }
 
-    public void openSession(@Nullable String sessionToken) throws FreeboxException {
-        WebSocketClient client = new WebSocketClient(apiHandler.getHttpClient());
-        URI uri = getUriBuilder().scheme(getUriBuilder().build().getScheme().contains("s") ? "wss" : "ws").build();
-        ClientUpgradeRequest request = new ClientUpgradeRequest();
-        request.setHeader(ApiHandler.AUTH_HEADER, sessionToken);
+    public void openSession(@Nullable String sessionToken, int reconnectInterval) {
+        if (reconnectInterval > 0) {
+            URI uri = getUriBuilder().scheme(getUriBuilder().build().getScheme().contains("s") ? "wss" : "ws").build();
+            ClientUpgradeRequest request = new ClientUpgradeRequest();
+            request.setHeader(ApiHandler.AUTH_HEADER, sessionToken);
+
+            stopReconnect();
+            reconnectJob = Optional.of(scheduler.scheduleWithFixedDelay(() -> {
+                try {
+                    closeSession();
+                    client.start();
+                    client.connect(this, uri, request);
+                    // Update listeners in case we would have lost data while disconnecting / reconnecting
+                    listeners.values().forEach(host -> host
+                            .handleCommand(new ChannelUID(host.getThing().getUID(), REACHABLE), RefreshType.REFRESH));
+                    logger.debug("Websocket manager connected to {}", uri);
+                } catch (Exception e) {
+                    logger.warn("Error connecting websocket client: {}", e.getMessage());
+                }
+            }, 0, reconnectInterval, TimeUnit.MINUTES));
+        }
+    }
+
+    private void stopReconnect() {
+        reconnectJob.ifPresent(job -> job.cancel(true));
+        reconnectJob = Optional.empty();
+    }
 
+    public void dispose() {
+        stopReconnect();
         try {
-            client.start();
-            client.connect(this, uri, request);
+            client.stop();
```

Comment:
Looks inconsistent, you will have more starts (every X minutes) than stops.
Either the stop or the start is not at the right place.

Reply:
@clinique : did you see my comment?

Reply:
What I mean is that you have a threadscheduled every X that start the client. But before you start it, you are not stopping it.

Reply:
Done